src/main/java/org/broadinstitute/hellbender/utils/codecs/sampileup/SAMPileupCodec.java

```diff
@@ -0,0 +1,343 @@
+ *     seq1 279 C 23  A..T,,.,.,...,,,.,..... ;75&<<<<<<<<<=<<<9<<:<<
+ * </pre>
+ * <p>
+ *     See the <a href="http://samtools.sourceforge.net/pileup.shtml">Pileup format documentation</a> for more details.
+ * </p>
+ *
+ * <h4>Consensus pileup: 10/13-column format</h4>
+ * <p>The "consensus" or extended pileup consists of the following:
+ *  <ul>
+ *      <li>original 6 columns as described above</li>
+ *      <li>4 extra columns representing consensus values (consensus base, consensus quality, variant quality and maximum mapping quality of the
+ * reads covering the sites) for all sites, inserted before the bases and quality strings</li>
+ *      <li>3 extra columns indicating counts of reads supporting indels (just for indel sites)</li>
+ *  </ul>
+ * </p>
+ * <h4>Example of consensus pileup for SNP or non-variant sites</h4>
+ * <pre>
+ *     seq1  60  T  T  66  0  99  13  ...........^~.^~.   9<<55<;<<<<<<
+ *     seq1  61  G  G  72  0  99  15  .............^~.^y. (;975&;<<<<<<<<
+ *     seq1  62  T  T  72  0  99  15  .$..............    <;;,55;<<<<<<<<
+ *     seq1  63  G  G  72  0  99  15  .$.............^~.  4;2;<7:+<<<<<<<
+ *     seq1  64  G  G  69  0  99  14  ..............  9+5<;;;<<<<<<<
+ *     seq1  65  A  A  69  0  99  14  .$............. <5-2<;;<<<<<<;
+ *     seq1  66  C  C  66  0  99  13  .............   &*<;;<<<<<<8<
+ *     seq1  67  C  C  69  0  99  14  .............^~.    ,75<.4<<<<<-<<
+ *     seq1  68  C  C  69  0  99  14  ..............  576<;7<<<<<8<< *
+ * </pre>
+ *
+ * <h4>Example of consensus pileup for indels</h4>
+ * <pre>
+ *     Escherichia_coli_K12	3995037	*	*\/*	430	0	37	144	*	+A	143	1	0
+ *     Escherichia_coli_K12	3995279	*	*\/*	202	0	36	68	*	+A	67	1	0
+ *     Escherichia_coli_K12	3995281	*	*\/*	239	0	36	67	*	-CG	66	1	0
+ * </pre>
+ * <p>
+ *     See <a href="http://samtools.sourceforge.net/cns0.shtml/">Consensus pileup format (deprecated)</a> for more details.
+ * </p>
+ *
+ * <h3>Caveat</h3>
+ * <p>Handling of indels is questionable at the moment. Proceed with care.</p>
+ *
+ *
+ * @author Matt Hanna, Geraldine VdAuwera
+ * @since 2014
+ */
+public class SAMPileupCodec extends AsciiFeatureCodec<SAMPileupFeature> {
+    // number of tokens expected (6 or 10 are valid, anything else is wrong)
+    private static final int basicTokenCount = 6;
+    private static final int consensusSNPTokenCount = 10;
+    private static final int consensusIndelTokenCount = 13;
+    private static final char fldDelim = '\t';
+    // allocate once and don't ever bother creating them again:
+    private static final String baseA = "A";
+    private static final String baseC = "C";
+    private static final String baseG = "G";
+    private static final String baseT = "T";
+    private static final String emptyStr = ""; // we will use this for "reference" allele in insertions
+
+    // codec file extension
+    protected static final String FILE_EXT = "samp";
```

Comment:
I removed "samp" (breaking compatibility with GATK3) and add the recomended ones ("pileup" and "mpileup")